### PR TITLE
Feature/mac os shortcuts

### PR DESCRIPTION
**Pull request by zephylac**
This feature add support for Mac.
Before this the modifier key was Control even on MacOS. 
With this fix, windows and linux will have Control as modifierKey whereas MacOS will have cmd which is more natural.

I decided to not add this to the configuration file because we wouldn't be compatible to each architecture or we would need to add modifier key for each platform.

I'm up for any suggestion to improve this.

### CODE REVIEW

File: text_model.go [hunk 1]
```diff
@@ -41,24 +41,24 @@ func (state *textModel) addChar(char []rune) {
...
 	if mods != ModShift {
 		state.selectionExtent = state.selectionBase
 	}
 	state.notifyState()
 }
 
-func (state *textModel) MoveCursorLeft(mods int) {
-	if mods == ModShiftControl || mods == ModControl {
+func (state *textModel) MoveCursorLeft(mods int, modifierKey int) {
+	if mods == ModShiftControl || mods == modifierKey {
```
Comment (Drakirus) [inline text_model.go]: Regression

File: text_model.go [hunk 2]
```diff
@@ -69,15 +69,15 @@ func (state *textModel) MoveCursorLeft(mods int) {
...
 		state.selectionExtent = state.selectionBase
 	}
 
 	state.notifyState()
 }
 
-func (state *textModel) MoveCursorRight(mods int) {
-	if mods == ModShiftControl || mods == ModControl {
+func (state *textModel) MoveCursorRight(mods int, modifierKey int) {
+	if mods == ModShiftControl || mods == modifierKey {
```
Comment (Drakirus) [inline text_model.go]: Regression

**Review by pchampio — DISMISSED** [review]
`ModShiftControl` is still hard coded.
Since the `modifierKey` is not necessarily `Control`, the change must take into account the new `modifierKey` in some statements.

It might me the time to move away from passing `int(mods)` into each `state` functions.
I would rather have a Boolean saying which modifier is active or not.
i.e.`state.MoveCursorLeft(true, false)` for CursorLeft + shift + no-ctrl/meta

**Comment by Drakirus** [comment]
https://transfer.sh/u2ct3/flutter.gif

Regression:  ctrl-Left/Right isn't working anymore.
Regression: when the key Left is pressed it send the cursor to the start of the input.

**Review by pchampio — CHANGES_REQUESTED** [review]
https://transfer.sh/u2ct3/flutter.gif

Regression:  ctrl-Left/Right isn't working anymore.
Regression: when the key Left is pressed it send the cursor to the start of the input.

**Comment by GeertJohan** [comment]
Have the regressions been resolved?

**Comment by GeertJohan** [comment]
I think using runtime.GOOS is good as well. I'll test this PR tomorow evening on linux.

**Comment by GeertJohan** [comment]
I've just merged master into this branch to make it compile again with the moved repo.

I did some testing. This still is this case on my linux machine:
> Regression: ctrl-Left/Right isn't working anymore.

The other regression has indeed been solved for linux.

**Comment by zephylac** [comment]
I'll try to look into it ASAP

**Pull request merged**